editor/components/block-drop-zone/index.js

```diff
@@ -105,6 +105,16 @@ class BlockDropZone extends Component {
 }
 
 export default compose(
+	withSelect( ( select ) => {
+		const { canInsertBlockType, getBlockName, getEditorSettings } = select( 'core/editor' );
+		const { templateLock } = getEditorSettings();
+
+		return {
+			isLocked: !! templateLock,
+			canInsertBlockType,
+			getBlockName,
+		};
+	} ),
 	withDispatch( ( dispatch, ownProps ) => {
```

Comment:
Regardless to what you decide in response to `DropZone` component, I wanted to leave a note about a way how you can avoid passing down selectors as props. A while back we have added 3rd param `registry` to `withDispatch` method which allows you to get access to the same selectors without exposing them as props. There is an example added in the docs: https://github.com/WordPress/gutenberg/tree/master/packages/data#withdispatch-mapdispatchtoprops-function--function (2nd example).

In this case, it would be something like the following:
```js
withDispatch( ( dispatch, ownProps, { select } ) => {
    // ...
    moveBlockToPosition( uid, fromRootUID, index ) {
        const { rootUID, layout } = ownProps;
        const { canInsertBlockType, getBlockName } = select( 'core/editor' );
        // and the rest as before
    }
}